includes/payments/functions.php

```diff
@@ -1225,6 +1224,34 @@ function edd_get_next_payment_number() {
 }
 
 /**
+ * Given a prefix, number and postfix, return the payment number,  incremented
+ *
+ * @since  2.4
+ * @param  string $prefix  The Payment Number Prefix
+ * @param  string $number  The formatted Current Number to increment
+ * @param  string $postfix The Payment Number Postfix
+ * @return string          The new formatted Payment number
+ */
+function edd_increment_payment_number( $prefix, $number, $postfix ) {
+
+	// Remove prefix and postfix
+	$number = str_replace( $prefix, '', $number );
+	$number = str_replace( $postfix, '', $number );
+
+	// Ensure it's a whole number
+	$number = intval( $number );
+
+	// Increment the payment number
+	$number++;
+
+	// Re-add the prefix and postfix
+	$number = $prefix . $number . $postfix;
+
+	return $number;
```

Comment:
There should be a filter here, so extensions can come in and say not to increment. Specific use case that comes to mind is in the EU, some countries cannot increment numbers on pending, they have to do it on completed. Not sure how the extension would work, but a filter can't kill us